File: Projects/IsPerfectSquare.py
import math
class IsPerfectSquare :
    def main ( self , args ) :
        print ( self . isPerfectSquare ( 0 ) == True ) 
        print ( self . isPerfectSquare ( 1 ) == True ) 
        print ( self . isPerfectSquare ( 16 ) == True ) 
        print ( self . isPerfectSquare ( 1234 * 1234 ) == True ) 
        print ( self . isPerfectSquare ( 15 ) == False ) 
        print ( self . isPerfectSquare ( - 16 ) == False ) 

    def isPerfectSquare ( self , n ) :
        if ( n < 0 ) :
            
            return False 

        x = math . sqrt ( n ) 
        if ( x * x == n ) :
            
            return True 

        return False 




IsPerfectSquare().main('')

File: Projects/IsPerfectSquare.java


public class IsPerfectSquare {

   public static void main(String[] args) {
        System.out.println(isPerfectSquare(0) == true);
    		System.out.println(isPerfectSquare(1) == true);
    		System.out.println(isPerfectSquare(16) == true);
    		System.out.println(isPerfectSquare(1234*1234) == true);
    		System.out.println(isPerfectSquare(15) == false);
    		System.out.println(isPerfectSquare(-16) == false);
   }

	public static boolean isPerfectSquare(int n){
		if (n < 0){
			return false;
		}
		double x = Math.sqrt(n);
		if (x*x == n){
			return true;
		}
		return false;
	}
}
        







